parser_library/src/processing/preprocessor.h

```diff
@@ -100,8 +113,13 @@ class preprocessor
     static void do_highlighting(
         const semantics::preprocessor_statement_si& stmt, semantics::source_info_processor& src_proc);
 
+    void store_included_member(included_member_details details);
+    void store_included_members(std::vector<included_member_details> details);
+    void capture_included_members(preprocessor& preproc);
+
 private:
     std::vector<std::shared_ptr<semantics::preprocessor_statement_si>> m_statements;
+    std::vector<included_member_details> m_inc_members;
```

Comment:
if the pointer to `text` content appears anywhere it may get invalidated when the vector reallocates; may not be an issues if the pointers are used only after the vector's content "stops moving".

The usage in `finish_preprocessor` looks fine to me. So if that's the only use, no change is needed.

Reply:
Hmm, you're right. Changed to `unique_ptr` to be safe.